Yesod/Auth/OAuth2.hs

```diff
@@ -52,28 +54,46 @@ authOAuth2 name oauth getCreds = AuthPlugin name dispatch login
     where
         url = PluginR name ["callback"]
 
-        withCallback = do
+        withCallback csrfToken = do
             tm <- getRouteToParent
             render <- lift $ getUrlRender
-            return $ oauth { oauthCallback = Just $ encodeUtf8 $ render $ tm url }
+            let newEndpoint = oauthOAuthorizeEndpoint oauth `BS.append` "&state=" `BS.append` encodeUtf8 csrfToken
```

Comment:
I think if we can confirm all the services here (GitHub, Learn, Spotify, and sort of Google) support the state parameter, then let's not worry about it. We know GitHub does, I can look into Learn. Could you look into Spotify and Google?